File: Assets/Zombie.js
#pragma strict

var speed = 0.1;
var animator : Animator;

function Start () {
	animator = GetComponent("Animator");
	
	// Workaround, the speed of the animation is randomly changed for 0.1 seconds so it seems like they start at different moments
	animator.speed = Random.Range(0,2000);
 	yield WaitForSeconds(0.1);	
    animator.speed = 1;
}

function Update () {
	updatePosition();
}

function updatePosition() {
	updateZ();
}

function updateZ() {
	GetComponent.<Rigidbody>().position.z += speed;
}

File: Assets/Zombie.cs
using UnityEngine;
using System.Collections;

public class Zombie : MonoBehaviour {

	public float speed = 0.1f;
	Animator animator;

	// Use this for initialization
	void Start () {
		animator = (Animator) GetComponent("Animator");
		// Workaround, the speed of the animation is randomly changed for 0.1 seconds so it seems like they start at different moments
		animator.speed = Random.Range(0,2000);
		StartCoroutine(wait ());
		animator.speed = 1;
	}
	
	// Update is called once per frame
	void Update () {
		updatePosition ();
	}

	void updatePosition() {
		updateZ();
	}
	
	void updateZ() {
		GetComponent<Rigidbody> ().position += new Vector3 (0, 0, speed);
	}

	IEnumerator wait() {
		yield return new WaitForSeconds (0.1f);
	}

	//MANAGE COLLISIONS
	void OnCollisionEnter(Collision collision) {
		if (collision.gameObject.name.Equals("Player")) {
			if (gameObject.name.Equals("BackZombie")) {	//zombie from the back (game over)
				Time.timeScale = 0;
			}
		}
	}
}
